scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,82 @@
+# 0010 - Separate Versioning Stream for Vulnerability Bundles
+
+- **Author(s):** J. Victor Martins <jvdm@sdf.org>
+- **Created:** [2024-07-22 Mon]
+
+## Status
+
+Status: Accepted
+
+Updates: [#0009](0009-scannerv4-read-manual-vulnerabilities.md)
+
+## Context
+
+A Vulnerability Bundle is an archive containing updates to vulnerability data in Scanner.  They are constantly updated by the Vulnerability Exporter (aka. Exporter), which runs periodically in GitHub's Workflows and uses ClairCore to fetch the security data to convert it to a JSON-based schema.  Then, the workflow bundles the data into a ZIP archive.  The schema is owned by ClairCore.
+
+A Scanner instance runs the Vulnerability Updater (aka. Updater) to fetch the bundle and use ClairCore to parse the schema and update the vulnerability database.  Suppose ClairCore changed the schema in a backward-incompatible manner, and that change is incorporate in the Scanner repository (a ClairCore version bump).  In that scenario, the next Exporter execution will update the bundle with the new schema.  Then, existing Scanner instances won't be able to consume the data.  A Scanner version bump to a version of ClairCore supporting the new schema would be required.
+
+To address that scenario, Scanner created one bundle per Stackrox release.  Each bundle is created with an Exporter running the same ClairCore version used to build the release, ensuring compatibility.  Bundles are served at `https://definitions.stackrox.io/v4/V/vulnerabilities.zip` where `V` will be a Stackrox release version such as "4.5.0".  Scanner uses its release version tagged during build time to determine which version to fetch from.  The versions to build bundles for are taken from a file named `scanner/updater/version/RELEASE_VERSION`.
```

Comment:
> Bundles are served at https://definitions.stackrox.io/v4/V/vulnerabilities.zip where V will be a Stackrox release version such as "4.5.0". Scanner uses its release version tagged during build time to determine which version to fetch from

nit: This is just the online bundles, since this is just contextual may be OK - for completeness may be worth calling out the offline bundles with different versioning semantics and location

Reply:
Update it.